core/src/main/java/edu/wpi/grip/core/http/GripServer.java

```diff
@@ -0,0 +1,404 @@
+                        he.sendResponseHeaders(STATUS_OK, rawResponseBody.length);
+                        he.getResponseBody().write(rawResponseBody);
+                        he.getResponseBody().close();
+                    } else {
+                        he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    }
+                    break;
+                case METHOD_POST:
+                    final byte[] bytes = readBytes(he);
+                    boolean success = true;
+                    for (PostHandler handler : postHandlers.get(path)) {
+                        success &= handler.convert(bytes);
+                    }
+                    if (success) {
+                        he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    } else {
+                        he.sendResponseHeaders(STATUS_INTERNAL_ERROR, NO_RESPONSE_LENGTH);
+                    }
+                    break;
+                case METHOD_OPTIONS:
+                    headers.set(HEADER_ALLOW, ALLOWED_METHODS);
+                    he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    break;
+                default:
+                    headers.set(HEADER_ALLOW, ALLOWED_METHODS);
+                    he.sendResponseHeaders(STATUS_METHOD_NOT_ALLOWED, NO_RESPONSE_LENGTH);
+                    break;
+            }
+        };
+    }
+
+    /**
+     * Adds a data supplier for data with the given name. Each name can only be
+     * associated with a single supplier at a time, so this will remove the previous
+     * supplier (if one exists).
+     *
+     * @param name     the name of the data
+     * @param supplier the supplier of the data
+     */
+    public void addDataSupplier(String name, Supplier<?> supplier) {
+        dataSuppliers.put(name, supplier);
+    }
+
+    /**
+     * Checks if the given name is associated with a supplier.
+     *
+     * @param name the name to check
+     * @return true if the name is associated with a supplier, false if not
+     */
+    public boolean hasDataSupplier(String name) {
+        return dataSuppliers.containsKey(name);
+    }
+
+    /**
+     * Removes the data supplier for data with the given name
+     *
+     * @param name the name of the data to remove
+     */
+    public void removeDataSupplier(String name) {
+        dataSuppliers.remove(name);
+    }
+
+    private String createJson(Map<String, List<String>> params) {
+        final Gson gson = new GsonBuilder()
+                .setPrettyPrinting()
+                .serializeSpecialFloatingPointValues()
+                .create();
+        final Map<String, Object> data = new HashMap<>();
+        dataSuppliers.entrySet()
+                .stream()
+                .filter(e -> params.isEmpty() || params.containsKey(e.getKey())) // send everything if nothing is specified
+                .forEach(e -> data.put(e.getKey(), e.getValue().get()));
```

Comment:
This can become:
```java
        final Map<String, Object> data = 
                dataSuppliers
                .entrySet()
                .stream()
                .filter(e -> params.isEmpty() || params.containsKey(e.getKey())) // send everything if nothing is specified
                .collect(Collectors.toMap(v -> v.getKey(), v -> v.getValue().get()));
```